openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/core/CoreMatrixAlgorithm.java

```diff
@@ -0,0 +1,538 @@
+            if(hasTurnWeighting && !isInORS(iter, currEdgeItem))
+                turnWeighting.setInORS(false);
+            edgeWeight = weighting.calcWeight(iter, swap, currEdgeItem.getOriginalEdge());
+            if(Double.isInfinite(edgeWeight))
+                continue;
+            double tmpWeight = edgeWeight + entryWeight;
+
+            MultiTreeSPEntryItem eeItem = adjEntry.getItem(i);
+            if (eeItem.getWeight() > tmpWeight) {
+                eeItem.setWeight(tmpWeight);
+                eeItem.setEdge(iter.getEdge());
+                eeItem.setOriginalEdge(EdgeIteratorStateHelper.getOriginalEdge(iter));
+                eeItem.setParent(currEdge);
+                eeItem.setUpdate(true);
+                addToQueue = true;
+            }
+            if(hasTurnWeighting)
+                turnWeighting.setInORS(true);
+        }
+
+        return addToQueue;
+    }
+
+    /**
+     * Update a target entry in the targetMap from an update entry. This is necessary to keep target results and running calculations separate
+     * @param update the new entry whose weights should update a target
+     */
+    private void updateTarget(AveragedMultiTreeSPEntry update) {
+        int nodeId = update.getAdjNode();
+        if(targetSet.contains(nodeId)) {
+            if (!targetMap.containsKey(nodeId)) {
+                AveragedMultiTreeSPEntry newTarget = new AveragedMultiTreeSPEntry(nodeId, EdgeIterator.NO_EDGE, Double.POSITIVE_INFINITY, true, null, update.getSize());
+                newTarget.setSubItemOriginalEdgeIds(EdgeIterator.NO_EDGE);
+                targetMap.put(nodeId, newTarget);
+            }
+            AveragedMultiTreeSPEntry target = targetMap.get(nodeId);
+            for (int i = 0; i < treeEntrySize; ++i) {
+                MultiTreeSPEntryItem targetItem = target.getItem(i);
+                double targetWeight = targetItem.getWeight();
+
+                MultiTreeSPEntryItem msptSubItem = update.getItem(i);
+                double updateWeight = msptSubItem.getWeight();
+
+                if (targetWeight > updateWeight) {
+                    targetItem.setWeight(updateWeight);
+                    targetItem.setEdge(msptSubItem.getEdge());
+                    targetItem.setOriginalEdge(msptSubItem.getOriginalEdge());
+                    targetItem.setParent(msptSubItem.getParent());
+                }
+            }
+        }
+    }
+
+    /**
+     * /
+     * /
+     * __________IN-CORE
+     * /
+     * /
+     **/
+
+    /**
+     * Create a Many to Many Dijkstra for the core and downwards phase and run it
+     */
+    private void runPhaseInsideCore() {
+        // Calculate all paths only inside core
+        DijkstraManyToMany algorithm = new DijkstraManyToMany(graph, chGraph, bestWeightMap, bestWeightMapCore, weighting, TraversalMode.NODE_BASED);
+
+        EdgeFilterSequence edgeFilterSequence = new EdgeFilterSequence();
+        edgeFilterSequence.add(this.additionalCoreEdgeFilter);
```

Comment:
Why is there a new instance of `EdgeFilterSequence` created, wouldn't simply passing the existing filter work? 🤔 
```
algorithm.setEdgeFilter(this.additionalCoreEdgeFilter);
```

Reply:
done